### PR TITLE
time_major parameter for ctc_loss op

### PR DESCRIPTION
**Pull request by igormq**
Just as dynamic_rnn and to be more regular across the TensorFlow's code I added a flag capable of transpose the input data to be in time_major, required by gen_ctc_ops._ctc_loss. Note that, instead of set time_major=False (as the most of TensorFlow data), it was set False to be compatible with previous ctc_loss function call (but I don't think that this is a good idea).

### CODE REVIEW

**Comment by ebrevdo** [comment]
Can you resolve conflicts?

**Comment by ebrevdo** [comment]
Jenkins, test this please

**Comment by igormq** [comment]
@ebrevdo What do you mean? I'm new on this.

**Comment by rmlarsen** [comment]
Can you sync you repo to the latest version of the master and resolve any merge conflicts that might occur?

**Comment by igormq** [comment]
@rmlarsen I did it, but I don't have any clue if a did it right. :)

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@igormq looks like you did it right :) 👍 
@ebrevdo would you mind reviewing this?

**Comment by ebrevdo** [comment]
May have you wait till Monday

**Comment by ebrevdo** [comment]
Err to wait

**Comment by danmane** [comment]
@ebrevdo Please take a look

File: tensorflow/python/ops/ctc_ops.py [hunk 1]
```diff
@@ -107,6 +112,13 @@ def ctc_loss(inputs, labels, sequence_length,
     preprocess_collapse_repeated: Boolean.  Default: False.
       If True, repeated labels are collapsed prior to the CTC calculation.
     ctc_merge_repeated: Boolean.  Default: True.
+    time_major: The shape format of the `inputs` Tensors.
+      If true, these `Tensors` must be shaped `[max_time, batch_size, num_classes]`.
```
Comment (ebrevdo) [inline tensorflow/python/ops/ctc_ops.py]: If `True`
below, If `False`

**Comment by ebrevdo** [comment]
Jenkins, test this please.

**Comment by ebrevdo** [comment]
One tiny docstring change; then it's good to merge.

**Comment by ebrevdo** [comment]
Comment here when you've modified the docstring.

**Comment by igormq** [comment]
@ebrevdo I've finished. 👍

**Comment by danmane** [comment]
@tensorflow-jenkins Test this please

**Comment by rmlarsen** [comment]
@andrewharp It looks like you can merge this.

**Pull request merged**